### ISSUE TITLE
Bug: Primitive unboxing breaks our handling of Map

### ISSUE THREAD

**Issue by lazaroclapp** [uber/NullAway#410]
The following code mistakenly reports a null dereference:

```
static void mapCheckWithPrimitiveUnboxing(int key) {
    Map<Integer, Object> m = new HashMap<>();
    if(m.containsKey(key)) {
      m.get(key).hashCode(); // NullAway errors here
    }
  }
```

While the following is correctly determined to be safe (note the type of the parameter `key`):

```
static void mapCheckWithPrimitiveUnboxing(Integer key) {
    Map<Integer, Object> m = new HashMap<>();
    if(m.containsKey(key)) {
      m.get(key).hashCode();
    }
  }
```

This seems to be some issue with how boxing and unboxing is handled in combination with our handling of Maps.

**Comment by lazaroclapp** [comment]
This is somewhat related to #398 and there might be a fix that encompasses both issues.

**Comment by lazaroclapp** [comment]
Closed by #413
